src/modules/universe/RadixUniverse.ts

```diff
@@ -224,7 +224,12 @@ export default class RadixUniverse {
                     })
 
                     nodeConnection.on('closed', () => {
-                        resolve(this.getNodeConnection(shard))
+                        this.getNodeConnection(shard)
+                            .then(connection => {
+                                resolve(connection)
+                            }).catch(e => {
+                                reject(e)
+                            })
```

Comment:
I think the new code is equivalent to the old code for all intents and purposes?

Reply:
Is it? Before, it would resolve even if there was a rejection from the recursive call